commit d93d9e40573a0c663815db81bac9b8719dee7ad6
Author: Mario Haustein <mario.haustein@hrz.tu-chemnitz.de>
Date:   Mon Dec 1 23:54:41 2025 +0100
Upstream: https://github.com/indilib/indi-3rdparty/pull/1214

    Link astroasis against libhidapi
    
    liboasisfilterwheel.bin and liboasisfocuser.bin references libhidapi
    functions. Thus the drivers need to be linked against libhidapi.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,6 +11,16 @@ find_package(ZLIB REQUIRED)
 find_package(USB1 REQUIRED)
 find_package(Threads REQUIRED)
 
+if(INDI_HIDAPILIB)
+    set(HIDAPILIB "")
+    message(STATUS "Using indi bundled hid library")
+else(INDI_HIDAPILIB)
+    find_package(hidapi REQUIRED)
+    add_definitions(-D_USE_SYSTEM_HIDAPILIB)
+    set(HIDAPILIB hidapi::libusb)
+    message(STATUS "Using system provided hid library")
+endif(INDI_HIDAPILIB)
+
 set(ASTROASIS_VERSION_MAJOR 1)
 set(ASTROASIS_VERSION_MINOR 1)
 
@@ -30,18 +40,18 @@ include(CMakeCommon)
 add_executable(indi_oasis_focuser ${CMAKE_CURRENT_SOURCE_DIR}/oasis_focuser.cpp)
 IF (APPLE)
 set(CMAKE_EXE_LINKER_FLAGS "-framework IOKit -framework CoreFoundation")
-target_link_libraries(indi_oasis_focuser ${INDI_LIBRARIES} ${ASTROASIS_LIBRARIES} ${LIBUSB_LIBRARIES})
+target_link_libraries(indi_oasis_focuser ${INDI_LIBRARIES} ${ASTROASIS_LIBRARIES} ${LIBUSB_LIBRARIES} ${HIDAPILIB})
 ELSE()
-target_link_libraries(indi_oasis_focuser ${INDI_LIBRARIES} ${ASTROASIS_LIBRARIES} ${USB1_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(indi_oasis_focuser ${INDI_LIBRARIES} ${ASTROASIS_LIBRARIES} ${USB1_LIBRARIES} ${HIDAPILIB} ${CMAKE_THREAD_LIBS_INIT})
 ENDIF()
 
 ########### indi_oasis_filter_wheel ###########
 add_executable(indi_oasis_filter_wheel ${CMAKE_CURRENT_SOURCE_DIR}/oasis_filter_wheel.cpp)
 IF (APPLE)
 set(CMAKE_EXE_LINKER_FLAGS "-framework IOKit -framework CoreFoundation")
-target_link_libraries(indi_oasis_filter_wheel ${INDI_LIBRARIES} ${ASTROASIS_LIBRARIES} ${LIBUSB_LIBRARIES})
+target_link_libraries(indi_oasis_filter_wheel ${INDI_LIBRARIES} ${ASTROASIS_LIBRARIES} ${LIBUSB_LIBRARIES} ${HIDAPILIB})
 ELSE()
-target_link_libraries(indi_oasis_filter_wheel ${INDI_LIBRARIES} ${ASTROASIS_LIBRARIES} ${USB1_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(indi_oasis_filter_wheel ${INDI_LIBRARIES} ${ASTROASIS_LIBRARIES} ${USB1_LIBRARIES} ${HIDAPILIB} ${CMAKE_THREAD_LIBS_INIT})
 ENDIF()
 
 #####################################
